### PR TITLE
Header / Footer based on Experience Fragments

### LINKED ISSUE #174: [RTC] Header/Footer Setup
User Story:
As a Front End Developer, I would like to be able quickly alter or enhance the visual style my header/footer via a combination of experience fragment, style system.

Acceptance Criteria:
1. The header/footer should be consistent across a website
2. The header/footer should work with MSM based live/language copies
3. The header/footer should allow for external JS libraries (ie.  munchkin.js for marketo)
4. The header/footer should be editable by a FED, but locked to content authors.
5. The header/footer should allow for customer specific HTML fragments (ie.  copy/paste HTML fragment source)
6. The header/footer should work within AEM's responsive grid

### PR DESCRIPTION
**Pull request by bpauli**
- add experience fragments for header and footer section and include it to the provided content-page-template
- experience fragments provide the same language structure as the content section of the page

fixes #174

### CODE REVIEW

File: src/main/archetype/ui.content/src/main/content/jcr_root/content/experience-fragments/__contentFolderName__/language-masters/en/.content.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    cq:adobeTargetExportFormat="html"
+    jcr:primaryType="sling:OrderedFolder"
+    jcr:title="${language_country.split('_').get(0)}">
```
Comment (jckautzmann) [inline src/main/archetype/ui.content/src/main/content/jcr_root/content/experience-fragments/__contentFolderName__/language-masters/en/.content.xml]: Isn't the title = "en" instead as the path is ".../language-masters/en"?

File: src/main/archetype/ui.content/src/main/content/jcr_root/content/experience-fragments/__contentFolderName__/language-masters/en/site/footer/.content.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="cq:Page">
+    <jcr:content
+        cq:tags="[]"
+        cq:template="/libs/cq/experience-fragments/components/experiencefragment/template"
+        jcr:primaryType="cq:PageContent"
+        jcr:title="Footer"
+        sling:resourceType="cq/experience-fragments/components/experiencefragment"/>
```
Comment (jckautzmann) [inline src/main/archetype/ui.content/src/main/content/jcr_root/content/experience-fragments/__contentFolderName__/language-masters/en/site/footer/.content.xml]: shouldn't it point to the XF Core Component instead?

File: src/main/archetype/ui.content/src/main/content/jcr_root/content/experience-fragments/__contentFolderName__/language-masters/en/site/header/.content.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="cq:Page">
+    <jcr:content
+        cq:tags="[]"
+        cq:template="/libs/cq/experience-fragments/components/experiencefragment/template"
+        jcr:primaryType="cq:PageContent"
+        jcr:title="Header"
+        sling:resourceType="cq/experience-fragments/components/experiencefragment"/>
```
Comment (jckautzmann) [inline src/main/archetype/ui.content/src/main/content/jcr_root/content/experience-fragments/__contentFolderName__/language-masters/en/site/header/.content.xml]: shouldn't it point to the XF Core Component instead?

File: src/main/archetype/ui.content/src/main/content/jcr_root/content/experience-fragments/__contentFolderName__/language-masters/en/.content.xml (hunk 1 above)
Comment (bpauli) [reply src/main/archetype/ui.content/src/main/content/jcr_root/content/experience-fragments/__contentFolderName__/language-masters/en/.content.xml]: The path is moved according to the input of the users input. This is done in the post-generation-groovy script in https://github.com/adobe/aem-project-archetype/pull/252/files#diff-346cccad6cc3468e44003d42921faec1R50

File: src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appsFolderName__/components/form/button/.content.xml [hunk 4]
```diff
@@ -3,4 +3,4 @@
     jcr:primaryType="cq:Component"
     jcr:title="Form Button"
     sling:resourceSuperType="core/wcm/components/form/button/v2/button"
-    componentGroup="${componentGroupName} Form"/>
+    componentGroup="${componentGroupName}.Content Form"/>
```
Comment (richardhand) [inline src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appsFolderName__/components/form/button/.content.xml]: shouldn't these be `${componentGraoupName}.form`?

File: src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appsFolderName__/components/content/accordion/.content.xml [hunk 5]
```diff
@@ -4,4 +4,4 @@
     jcr:primaryType="cq:Component"
     jcr:title="Accordion"
     sling:resourceSuperType="core/wcm/components/accordion/v1/accordion"
-    componentGroup="${componentGroupName}"/>
+    componentGroup="${componentGroupName}.Content"/>
```
Comment (richardhand) [inline src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appsFolderName__/components/content/accordion/.content.xml]: Would use lower case here `${componentGroupName}.content`

File: src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/policies/.content.xml [hunk 6]
```diff
@@ -12,7 +12,7 @@
                         jcr:primaryType="nt:unstructured"
                         jcr:title="${siteName} Default"
                         sling:resourceType="wcm/core/components/policy/policy"
-                        components="[group:${componentGroupName},/apps/${appsFolderName}/components/form/container]"
+                        components="[group:${componentGroupName}.Content,/apps/${appsFolderName}/components/form/container,group:${componentGroupName}.Structure]"
```
Comment (richardhand) [inline src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/policies/.content.xml]: `group:${componentGroupName}.Content`,`group:${componentGroupName}.Form,`

File: src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/policies/.content.xml [hunk 7]
```diff
@@ -46,6 +75,224 @@
...
+                    </policy_xfm>
+                </title>
+                <experiencefragment jcr:primaryType="nt:unstructured">
+                    <policy_header
+                        jcr:title="${siteName} Experience Fragment - Header"
+                        sling:resourceType="wcm/core/components/policy/policy">
+                        <jcr:content jcr:primaryType="nt:unstructured"/>
+                    </policy_header>
+                    <policy_footer
+                    jcr:title="${siteName} Experience Fragment - Footer"
```
Comment (richardhand) [inline src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/policies/.content.xml]: trivial indent format missing here

File: src/main/archetype/ui.content/src/main/content/jcr_root/content/experience-fragments/__contentFolderName__/language-masters/en/site/footer/master/.content.xml [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+                            </cq:responsive>
+                        </button_instagram>
+                    </responsivegrid>
+                    <separator
+                        jcr:primaryType="nt:unstructured"
+                        sling:resourceType="${appsFolderName}/components/content/separator"/>
+                    <text
+                        jcr:primaryType="nt:unstructured"
+                        sling:resourceType="${appsFolderName}/components/content/text"
+                        text="&lt;p>Ⓒ 2019, ${siteName}.&lt;/p>&#xd;&#xa;"
```
Comment (richardhand) [inline src/main/archetype/ui.content/src/main/content/jcr_root/content/experience-fragments/__contentFolderName__/language-masters/en/site/footer/master/.content.xml]: can we somehow inject the year here so we don't have to update it?

File: src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appsFolderName__/components/form/button/.content.xml (hunk 4 above)
Comment (bpauli) [reply src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appsFolderName__/components/form/button/.content.xml]: The goal for naming is to align it with the WKND tutorial (see for example https://github.com/adobe/aem-guides-wknd/blob/develop/ui.content/src/main/content/jcr_root/conf/wknd/settings/wcm/policies/.content.xml#L106)

Comment (bpauli) [reply src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appsFolderName__/components/form/button/.content.xml]: see naming comment above

File: src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/policies/.content.xml (hunk 6 above)
Comment (bpauli) [reply src/main/archetype/ui.content/src/main/content/jcr_root/conf/__confFolderName__/settings/wcm/policies/.content.xml]: Only the form container should be available for the root repsonsivegrid.

File: src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appsFolderName__/components/form/button/.content.xml (hunk 4 above)
Comment (richardhand) [reply src/main/archetype/ui.apps/src/main/content/jcr_root/apps/__appsFolderName__/components/form/button/.content.xml]: ok, makes sense to align, but the grouping in WKND is strange to me, given the project structure:
```
/components/form
/components/content
/components/structure
```

The `myComponentGroup.Content Form` naming is a little confusing to me, if it's really a subgroup shouldn't it be `myComponentGroup.Content.Form`.

**Pull request merged**